sdks/python/apache_beam/runners/direct/sdf_direct_runner_test.py

```diff
@@ -0,0 +1,237 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Unit tests for SDF implementation for DirectRunner."""
+
+import logging
+import os
+import unittest
+
+import apache_beam as beam
+from apache_beam import DoFn
+from apache_beam.io import filebasedsource_test
+from apache_beam.io.restriction_trackers import OffsetRestrictionTracker
+from apache_beam.testing.test_pipeline import TestPipeline
+from apache_beam.testing.util import assert_that
+from apache_beam.testing.util import equal_to
+from apache_beam.transforms.core import ProcessContinuation
+from apache_beam.transforms.core import RestrictionProvider
+from apache_beam.transforms.trigger import AccumulationMode
+from apache_beam.transforms.window import SlidingWindows
+from apache_beam.transforms.window import TimestampedValue
+
+
+class ReadFilesProvider(RestrictionProvider):
```

Comment:
Hm, why not use the simpler example of generating a range of numbers? (this one's fine too)